src/ui/FileViewer/hooks/useKeyDown.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback, useLayoutEffect } from 'react';
+import { noop } from '../../../utils/utils';
+import { FileViewerComponentProps, ViewerTypes } from '../types';
+
+export function useKeyDown({ props, ref }: {
+  props: FileViewerComponentProps,
+  ref: React.RefObject<HTMLDivElement>;
+}): {
+  onKeyDown: React.KeyboardEventHandler<HTMLDivElement>;
+} {
+  const { onClose } = props;
+  let onClickLeft = noop;
+  let onClickRight = noop;
+  if (props.viewerType === ViewerTypes.MULTI) {
+    onClickLeft = props.onClickLeft;
+    onClickRight = props.onClickRight;
+  }
```

Comment:
@liamcho  before we put 
```
if (props.viewerType === ViewerTypes.MULTI) {
  useEffect...
  useCallback...
}
```
^^ this will break the rule of hooks 

Read -> https://react.dev/learn/state-a-components-memory#meet-your-first-hook

Reply:
nice workaround